### PR TITLE
Only precompute and cache indices in the OOP process.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
Followup to: 

https://github.com/dotnet/roslyn/pull/43636 and https://github.com/dotnet/roslyn/pull/43771

### CODE REVIEW

File: src/Features/Core/Portable/IncrementalCaches/SyntaxTreeInfoIncrementalAnalyzerProvider.cs [hunk 1]
```diff
@@ -35,11 +34,6 @@ public override Task AnalyzeSyntaxAsync(Document document, InvocationReasons rea
                     return Task.CompletedTask;
                 }
 
-                if (!RemoteFeatureOptions.ShouldComputeIndex(document.Project.Solution.Workspace))
```
Comment (tmat) [inline src/Features/Core/Portable/IncrementalCaches/SyntaxTreeInfoIncrementalAnalyzerProvider.cs:38]: >ShouldComputeIndex [](start = 42, length = 18)

Is this method still used?

Comment (CyrusNajmabadi) [reply src/Features/Core/Portable/IncrementalCaches/SyntaxTreeInfoIncrementalAnalyzerProvider.cs:38]: let me check.  i will remove if not!

File: src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_FindReferences_Current.cs [hunk 2]
```diff
@@ -27,12 +26,12 @@ public static partial class SymbolFinder
...
         {
-            Contract.ThrowIfNull(solution.GetOriginatingProjectId(symbol), WorkspacesResources.Symbols_project_could_not_be_found_in_the_provided_solution);
-
             using (Logger.LogBlock(FunctionId.FindReference, cancellationToken))
             {
+                var project = solution.GetOriginatingProject(symbol);
                 var client = await RemoteHostClient.TryGetClientAsync(solution.Workspace, cancellationToken).ConfigureAwait(false);
-                if (client != null)
+                if (client != null &&
+                    project != null)
```
Comment (tmat) [inline src/Workspaces/Core/Portable/FindSymbols/SymbolFinder_FindReferences_Current.cs:34]: >project != null [](start = 20, length = 15)

Nit: if project is null we don't need to fetch the client.

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder_Remote.cs [hunk 3]
```diff
@@ -25,16 +25,18 @@ internal static partial class DependentTypeFinder
         {
             using (Logger.LogBlock(functionId, cancellationToken))
             {
+                var project = solution.GetOriginatingProject(type);
                 var client = await RemoteHostClient.TryGetClientAsync(solution.Workspace, cancellationToken).ConfigureAwait(false);
-                if (client != null)
+                if (client != null &&
+                    project != null)
```
Comment (tmat) [inline src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder_Remote.cs:31]: > project != null [](start = 19, length = 16)

can return null immediately if project is null w/o fetching the client

File: src/Workspaces/Core/Portable/Rename/RenameLocations.cs [hunk 4]
```diff
@@ -125,22 +125,23 @@ internal sealed partial class RenameLocations
...
 
             cancellationToken.ThrowIfCancellationRequested();
 
             using (Logger.LogBlock(FunctionId.Renamer_FindRenameLocationsAsync, cancellationToken))
             {
+                var project = solution.GetOriginatingProject(symbol);
                 var client = await RemoteHostClient.TryGetClientAsync(solution.Workspace, cancellationToken).ConfigureAwait(false);
-                if (client != null)
+                if (client != null &&
+                    project != null)
```
Comment (tmat) [inline src/Workspaces/Core/Portable/Rename/RenameLocations.cs:136]: >project [](start = 20, length = 7)

The same question. Not using OOP if project == null?

File: src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder_Remote.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder_Remote.cs:31]: true.

**Pull request merged**